### ISSUE TITLE
[Linux] Install and configure .desktop file on Zotero launch

### ISSUE THREAD

**Issue by adomasven** [zotero/zotero#4691]
Many Linux applications do this. On running Zotero we should check if we're running from a user-writeable directory (installed in userspace and as such probably not via the .deb package), and if so, install the .desktop file into `~/.local/share/applications` with the correctly set path to the Zotero binary and then place the Zotero icon into `~/.local/share/icons`. Then the Icon entry can simply be `Icon=Zotero`.

**Comment by dstillman** [comment]
So the "installation" process would be putting the folder somewhere and running `./zotero` once from the command line?

**Comment by adomasven** [comment]
Well installation is putting the folder somewhere. This is setting up a .desktop entry so that it's listed in the application manager of user's choice.

**Comment by dstillman** [comment]
Installation as in what we'll say on the [Installation page](https://www.zotero.org/support/installation#linux). I'm just saying that instead of 2 and 3 on there, we'll just say to run `./zotero` once.

**Comment by dstillman** [comment]
What are some examples of apps that do this?

**Comment by adomasven** [comment]
Intellij IDEs do this. Wine creates entries there for Windows software that you run/install. I had a bunch of entries from all kinds of software in `.local/share/applications` but I recently removed them as part of general cleanup of buildup from the last 8 years. It's not uncommon and the intended way for userspace-installed-software to make itself available to launchers.

**Comment by cadadr** [comment]
In 20 years on Linux I have never encountered this method of installation. Nevertheless I think it's quite nice.

It might be a nice idea to (1) make sure it respects whatever `XDG_...` variable governs `~/.local/share/applications`, (2) more importantly, instead of doing it automatically, to ask for user's confirmation via a dialog in the first run, and (3) perhaps a way to pick an alternate directory, because you can have more than one directory where these .desktop files go, by adding extra directories to `XDG_DATA_DIRS`. For example, I used to have a separate one for my personal .desktop files set in it.

I admit (3) is a bit extra and you can probably expect a user knowledgeable enough to know about `XDG_DATA_DIRS` to be able to manage it themselves. But I would really like an option to say no to the installation of the file by Zotero itself because it might end up, in a few scenarios, adding an extra Zotero entry to the application menu, which might then have problematic results if the two different installations diverge somehow, e.g. in their version.

**Comment by adomasven** [comment]
Right, it looks like per XDG spec we should use `XDG_DATA_HOME` and when that's undefined `~/.local/share` for these paths. For most users installed in userspace its probably the correct thing to do to just install the file, but we could make it a menu option to add or remove it to prevent duplicates.

**Comment by adomasven** [comment]
Superseded by https://github.com/zotero/zotero/issues/5014

**Comment by dstillman** [comment]
After further consideration, this option makes more sense.
